test/stablex/batch_exchange.js

```diff
@@ -2044,14 +2044,34 @@ contract("BatchExchange", async accounts => {
   })
   describe("Large Examples", () => {
     it("ensures hard gas limit on largest possible ring trade ", async () => {
-      const batchExchange = await setupGenericStableX(25)
-      const sixPointFiveMillion = 6500000
-      await makeDeposits(batchExchange, accounts, longRingTrade.deposits)
+      const batchExchange = await setupGenericStableX(30)
+      const fivePointFiveMillion = 5500000
+
+      const tradeExample = largeRing30
+
+      // Deposit double sufficient amount and immediately request withdraw for half
+      await makeDeposits(batchExchange, accounts, tradeExample.deposits, 2)
+      for (const deposit of tradeExample.deposits) {
+        const tokenAddress = await batchExchange.tokenIdToAddressMap.call(deposit.token)
+        await batchExchange.requestWithdraw(tokenAddress, deposit.amount, { from: accounts[deposit.user] })
+      }
+      await closeAuction(batchExchange)
+
       const batchId = (await batchExchange.getCurrentBatchId.call()).toNumber()
-      const orderIds = await placeOrders(batchExchange, accounts, longRingTrade.orders, batchId + 1)
+      const orderIds = await placeOrders(batchExchange, accounts, tradeExample.orders, batchId + 1)
       await closeAuction(batchExchange)
 
-      const solution = solutionSubmissionParams(longRingTrade.solutions[0], accounts, orderIds)
+      // Ensure that first 30 orders have valid withdraw requests.
+      for (const deposit of tradeExample.deposits.slice(0, 30)) {
+        const tokenAddress = await batchExchange.tokenIdToAddressMap.call(deposit.token)
```

Comment:
Here, you are checking that the withdrawRequest is available for the selltoken, but I guess we need to have it for the bought tokens.

Reply:
Oh no. Really? Why? Nobody has a balance of buy token to withdraw anyway.

Reply:
You are right. The withdraw is only blocked when we add balance, so we need to request the withdraws for the buy not the sell token. 

Great catch.

Reply:
@bh2smith. They will after the first solution is applied. That’s why we need to block the withdraw because that balance is not yet final and thus shouldn’t be withdrawn until the next batch.

Reply:
How can all this stuff happen in one batch. I will revert and try again

Reply:
How can all this stuff happen in one batch. I will revert and try again